master/buildbot/process/workerforbuilder.py

```diff
@@ -16,14 +16,16 @@
 
 from twisted.internet import defer
 from twisted.python import log
+from twisted.python.constants import NamedConstant
+from twisted.python.constants import Names
 
-(ATTACHING,  # worker attached, still checking hostinfo/etc
- IDLE,  # idle, available for use
- PINGING,  # build about to start, making sure it is still alive
- BUILDING,  # build is running
- LATENT,  # latent worker is not substantiated; similar to idle
- SUBSTANTIATING,
- ) = range(6)
+
+class States(Names):
```

Comment:
While the source code shows interesting use of metaclass programming, NamedConstant and Names could use a bit more documentation :)
for others reviewers

```
>>> States.ATTACHING
<States=ATTACHING>
```